### PR TITLE
Added ability to change color maps

### PR DESCRIPTION
**Pull request by HanKruiger**
The `ColorMap` and `ColorLegend` classes are now used in `HeatMap` and `HeatMapArray`. It would be nice to also use these in `Histogram2D`, but this is more challenging, as it should then also support selection in the legend.

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
We'd like to support selection in the legend for these too!

File: web/src/main/webapp/heatMap.ts [hunk 1]
```diff
@@ -391,10 +326,28 @@ export class HeatMapView extends RemoteTableObjectView {
...
+            .attr("data-val", d => d.v)
             .attr("width", this.pointWidth)
             .attr("height", this.pointHeight)
             .style("stroke-width", 0)
-            .style("fill", d => this.color(d.v, max));
+            .style("fill", d => this.colorMap.apply(d.v));
+
+        this.colorLegend.redraw();
+        this.colorLegend.setColorMapChangeEventListener((cm) => {
+            this.chart.selectAll(".heatMapCell")
```
Comment (mbudiu-vmw) [inline web/src/main/webapp/heatMap.ts]: Can you turn this code into a method of 'this'?

File: web/src/main/webapp/menu.ts [hunk 2]
```diff
@@ -30,14 +30,16 @@ export class ContextMenu implements IHtmlElement {
         this.outer = document.createElement("table");
         this.outer.classList.add("dropdown");
         this.outer.classList.add("menu");
-        this.outer.onmouseleave = () => {this.clear(); this.hide()};
+        // this.outer.onmouseleave = () => {this.clear(); this.hide()};
```
Comment (mbudiu-vmw) [inline web/src/main/webapp/menu.ts]: Please delete the unused code.

**Comment by HanKruiger** [comment]
I've pushed the changes you requested!

Selection in the color legend can be done similarly as how the new colormap selection is done. With an `addSelectionListener` method that triggers the listener when a selection is made in the color legend. But this is future work.

**Pull request merged**